### DISCUSSION TITLE
How to create Awesome Repositories

### DISCUSSION THREAD

**Discussion by ghost** [community/community#103159] [Programming Help]
## Overview of my problem
So actually I have somewhat ok repositories for about 2 years. But my repos Don't get that reach or even contributions, I know my repositories are not enough. But now I wanna learn how to create 🚀🔥 Repositories. Ofcourse I know how to create a repo.

## My questions

1. How to create awesome repos that get more stars, contributions, and reach.
2. I have been using features like Discussions, topics, actions and others. I know about releases and tags but not used any.Am I missing any feature that can help me to do the first one/question.
3. How to come up with ideas for projects.

## Gratitude 
I thank all of you in advance.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by gil-son** [comment]
The purpose of the repository needs to be clarified so that other developers can identify the values of the project.

To achieve this, you have to showcase your project using the README.md file:

  1. Clearly state the purpose of the project.
  2. Outline the technologies and requirements.
  3. Provide instructions on how to execute it if necessary.
  4. Include illustrations.
  5. Review the ideas.

However, an amazing README.md is not enough. You also need to be able to 'sell your project.' For example, use platforms to give previews of your project, such as:

  - Creating videos on YouTube
  -  Sharing your project on servers in Discord
  - Creating posts on LinkedIn
  - Making it public on your portfolio

The consequences of that are:

- Increased access to your repository.
- More chance to receive stars.
- Improved chances of receiving contributions in yours projects.

**Reply by ghost** [reply]
Is there any particular github features that will help me in this

**Reply by ghost** [reply]
And also really thank you for suggesting these I will sure take this in account

**Reply by gil-son** [reply]
It's my pleasure

If you require some inspiration [look that](https://github.com/gil-son/aws)

**Reply by ghost** [reply]
That was awesome, I will sure follow it

**Reply by ghost** [reply]
Here is mine can you add insights about it

**Reply by ghost** [reply]
[Link](https://github.com/Yuheshpandian/369-CSS-EFFECTS)

**Reply by gil-son** [reply]
If it makes sense, consider marking the best answer
